pkg/nexus/nexus.go

```diff
@@ -53,53 +54,125 @@ func (n *BinaryDownloader) DownloadArtifact(c *config.MavenGav) (Deliverable, er
 	return deliverable, nil
 }
 
-func (n *NexusDownloader) DownloadArtifact(c *config.MavenGav) (Deliverable, error) {
-	resourceUrl, err := n.createURL(c)
-	deliverable := Deliverable{}
-	if err != nil {
-		return deliverable, errors.Wrapf(err, "Failed to create Nexus url for GAV %+v", c)
-	}
+func (n *NexusDownloader) DownloadArtifact(c *config.MavenGav, na *config.NexusAccess) (Deliverable, error) {
 
-	logrus.Debugf("Downloading artifact from %s", resourceUrl)
-	httpResponse, err := http.Get(resourceUrl)
-	if err != nil {
-		return deliverable, errors.Wrapf(err, "Failed to get artifact from Nexus %s", resourceUrl)
-	}
-	defer httpResponse.Body.Close()
-
-	if httpResponse.StatusCode != http.StatusOK {
-		return deliverable, errors.Errorf("Could not download artifact (Make sure you have deployed it!)"+
-			". Status code %s ", httpResponse.Status)
-	}
-
-	contentDisposition := httpResponse.Header.Get("content-disposition")
-
-	if len(contentDisposition) <= 0 {
-		return deliverable, errors.Errorf("No content-disposition in response header")
-	}
-
-	_, params, err := mime.ParseMediaType(contentDisposition)
-	if err != nil {
-		return deliverable, errors.Wrap(err, "Failed to parse content-disposition")
-	}
-	dir, err := ioutil.TempDir("", "package")
-	if err != nil {
-		return deliverable, errors.Wrap(err, "Failed to create directory for artifact")
-	}
-	fileName := filepath.Join(dir, params["filename"])
+	deliverable := Deliverable{}
 
-	fileCreated, err := os.Create(fileName)
+	// Detect API version
+	useNexus3 := false
+	resp, err := http.Get(n.baseUrl)
 	if err != nil {
-		return deliverable, errors.Wrap(err, "Failed to create artifact file")
+		logrus.Fatalf("Failed response from %s, error: %s", n.baseUrl, err)
+	} else {
+		rx := regexp.MustCompile(`Nexus/\d`)
+		nexusVer := rx.FindString(resp.Header.Get("Server"))
```

Comment:
Jeg tenker sånn umiddelbart at det er lettere å lese med

```
useNexus, _ := regexp.MatchString(`^Nexus/3\..*$`, resp.Header.Get("Server"))
```

Vet ikke hva andre syns om det..

Reply:
Det er mer kompakt og sikkert like greit å lese.  Jeg får ikke strengen direkte for logging (som jeg synes er praktisk i overgangsfasen med ulike Nexus-servere), men jeg kan logge den boolske variabelen istedet.  Endrer.